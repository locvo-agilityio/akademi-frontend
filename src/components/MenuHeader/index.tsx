import { Avatar, Button, Flex, HStack, Icon, Text } from '@chakra-ui/react';

// Icons
import { BellIcon, SettingIcon } from '../icons';

interface MenuHeaderProps {
  name: string;
  role: string;
  avatar?: string;
}

const MenuHeader = ({ name, role, avatar = '' }: MenuHeaderProps) => (
  <HStack gap={6}>
    <Flex gap={5}>
      <Button
        aria-label="Notification"
        w="60px"
        h="60px"
        bgColor="white"
        boxShadow="2xl"
        variant="icon"
      >
        <Icon as={BellIcon} boxSize={8} />
      </Button>

      <Button
        aria-label="Settings"
        w="60px"
        h="60px"
        bgColor="white"
        boxShadow="2xl"
        variant="icon"
      >
        <Icon as={SettingIcon} boxSize={8} />
      </Button>
    </Flex>

    <Flex direction="column">
      <Text fontWeight="bold" color="darkBlue">
        {name}.
      </Text>
      <Text as="span" color="gray.600" textAlign="right">
        {role}
      </Text>
    </Flex>

    <Avatar size="md" name={name} src={avatar} />
  </HStack>
);

export default MenuHeader;
