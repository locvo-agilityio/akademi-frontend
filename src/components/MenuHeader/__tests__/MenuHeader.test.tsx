import { render } from '@testing-library/react';

// Components
import MenuHeader from '..';

describe('MenuHeader component', () => {
  const mockProps = {
    name: 'John Doe',
    role: 'Admin',
  };

  it('Render correctly with default props', () => {
    const { container } = render(<MenuHeader {...mockProps} />);
    expect(container).toMatchSnapshot();
  });
});
