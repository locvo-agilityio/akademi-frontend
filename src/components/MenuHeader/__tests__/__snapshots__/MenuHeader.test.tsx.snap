// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuHeader component Render correctly with default props 1`] = `
<div>
  <div
    class="chakra-stack css-zpd0pq"
  >
    <div
      class="css-1j52cd2"
    >
      <button
        aria-label="Notification"
        class="chakra-button css-jqgc5n"
        type="button"
      >
        <svg
          class="chakra-icon css-rgiubw"
          fill="none"
          focusable="false"
          height="32"
          viewBox="0 0 26 32"
          width="26"
        >
          <path
            d="M22.3677 18.9391V9.86768C22.3677 4.70215 18.1655 0.5 13 0.5C7.83447 0.5 3.63232 4.70215 3.63232 9.86768V18.9397C1.96704 19.4224 0.738281 20.9544 0.738281 22.8374C0.738281 25.0386 2.5293 26.8296 4.73096 26.8296H8.37696V26.877C8.37695 29.4263 10.4507 31.5 13 31.5C15.5493 31.5 17.6231 29.4263 17.6231 26.8769V26.8296H21.2691C23.4707 26.8296 25.2617 25.0386 25.2617 22.7583C25.2617 20.9406 24.033 19.4198 22.3677 18.9391ZM6.63232 9.86768C6.63232 6.35645 9.48877 3.5 13 3.5C16.5112 3.5 19.3677 6.35645 19.3677 9.86768V18.7661H6.63232V9.86768ZM14.6231 26.8769C14.6231 27.772 13.895 28.5 13 28.5C12.105 28.5 11.377 27.772 11.377 26.8769V26.8296H14.623V26.8769H14.6231ZM21.269 23.8296H4.73096C4.1836 23.8296 3.73828 23.3843 3.73828 22.7583C3.73828 22.2114 4.18359 21.7661 4.73096 21.7661H21.2691C21.8164 21.7661 22.2617 22.2114 22.2617 22.8374C22.2617 23.3843 21.8164 23.8296 21.269 23.8296Z"
            fill="#A098AE"
          />
        </svg>
      </button>
      <button
        aria-label="Settings"
        class="chakra-button css-jqgc5n"
        type="button"
      >
        <svg
          class="chakra-icon css-rgiubw"
          fill="none"
          focusable="false"
          height="28"
          viewBox="0 0 28 28"
          width="28"
        >
          <path
            d="M10.2632 0.666748L9.41683 4.46623C8.93546 4.69787 8.47531 4.96466 8.03402 5.26571L4.32048 4.09644L0.583496 10.5704L3.3986 13.1485C3.28861 13.9649 3.33843 14.3673 3.3986 14.8516L0.583496 17.4298L4.32048 23.9037L8.03402 22.7345C8.47531 23.0355 8.93546 23.3023 9.41683 23.5339L10.2632 27.3334H17.7371L18.5835 23.5339C19.0649 23.3023 19.525 23.0355 19.9663 22.7345L23.6798 23.9037L27.4168 17.4298L24.6017 14.8516C24.6247 14.5682 24.6665 14.2846 24.6668 14.0001C24.6679 13.7069 24.6217 13.4109 24.6017 13.1485L27.4168 10.5704L23.6798 4.09644L19.9663 5.26571C19.525 4.96466 19.0649 4.69787 18.5835 4.46623L17.7371 0.666748H10.2632ZM12.4038 3.33341H15.5965L16.2554 6.29175L16.9168 6.55217C17.6651 6.84519 18.3645 7.24854 18.9923 7.75008L19.5496 8.19279L22.4403 7.28394L24.0366 10.0496L21.8022 12.099L21.909 12.8022C22.0347 13.5798 22.0103 14.4746 21.909 15.198L21.8022 15.9011L24.0366 17.9506L22.4403 20.7162L19.5496 19.8074L18.9923 20.2501C18.3645 20.7516 17.6651 21.155 16.9168 21.448L16.2554 21.7084L15.5965 24.6667H12.4038L11.745 21.7084L11.0835 21.448C10.3353 21.155 9.63581 20.7516 9.00797 20.2501L8.45068 19.8074L5.56006 20.7162L3.9637 17.9506L6.19808 15.9011L6.09131 15.198C5.96108 14.4047 5.98107 13.4968 6.09131 12.8022L6.19808 12.099L3.9637 10.0496L5.56006 7.28394L8.45068 8.19279L9.00797 7.75008C9.63581 7.24854 10.3353 6.84519 11.0835 6.55217L11.745 6.29175L12.4038 3.33341ZM14.0002 8.66675C11.0704 8.66675 8.66683 11.0704 8.66683 14.0001C8.66683 16.9298 11.0704 19.3334 14.0002 19.3334C16.9299 19.3334 19.3335 16.9298 19.3335 14.0001C19.3335 11.0704 16.9299 8.66675 14.0002 8.66675ZM14.0002 11.3334C15.4887 11.3334 16.6668 12.5115 16.6668 14.0001C16.6668 15.4886 15.4887 16.6667 14.0002 16.6667C12.5116 16.6667 11.3335 15.4886 11.3335 14.0001C11.3335 12.5115 12.5116 11.3334 14.0002 11.3334Z"
            fill="#A098AE"
          />
        </svg>
      </button>
    </div>
    <div
      class="css-j7qwjs"
    >
      <p
        class="chakra-text css-tbw3nr"
      >
        John Doe
        .
      </p>
      <span
        class="chakra-text css-r6hidi"
      >
        Admin
      </span>
    </div>
    <span
      class="chakra-avatar css-1ruhx4w"
    >
      <div
        aria-label="John Doe"
        class="chakra-avatar__initials css-0"
        role="img"
      >
        JD
      </div>
    </span>
  </div>
</div>
`;
